packages/batch-submitter/src/batch-submitter/batch-submitter.ts

```diff
@@ -223,6 +239,40 @@ export abstract class BatchSubmitter {
 
     this.logger.info('Received transaction receipt', { receipt })
     this.logger.info(successMessage)
+    this.bsMetrics.submissionGasUsed.observe(receipt.gasUsed.toNumber())
+    this.bsMetrics.submissionTimestamp.observe(Date.now())
     return receipt
   }
+
+  private _registerMetrics(metrics: Metrics): BatchSubmitterMetrics {
+    metrics.registry.clear()
+
+    return {
```

Comment:
@snario i'm not familiar yet with how registries work, but i have a feeling that the failing integration test might be due to registering the same metric so i added this line - is this standard practice or a hack? i could also only reset these specific metrics i'm registering here

Reply:
I don't think we should ever be registering the same metrics within the same process. It either indicates we ought to be mocking out `metrics` in the test or using different processes to test the behaviour of multiple batch submitters.

Reply:
i think it actually may be when we initialize both transaction and state batch submitters here:
https://github.com/ethereum-optimism/optimism/blob/ae1ac05d7032422a71caf25d16f6e548df5b8d7f/packages/batch-submitter/src/exec/run-batch-submitter.ts#L167-L213